### PR TITLE
[cli11] Update to version 1.9.0

### PR DESCRIPTION
**Pull request by LilyWangL**
Update cli11 to version 1.9.0. Fix the problem as below:
1. Remove `include(vcpkg_common_functions)`.
2. Disable build documentation for packages.
3. Add build dependency getst, this is new submodule. Related the source issue: https://github.com/CLIUtils/CLI11/issues/366

The related issue: #10121

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: ports/cli11/portfile.cmake [hunk 1]
```diff
@@ -14,6 +13,7 @@ vcpkg_configure_cmake(
     OPTIONS
         -DCLI11_TESTING=OFF
         -DCLI11_EXAMPLES=OFF
+        -DENABLE_DOCS=OFF
```
Comment (DailyShana) [inline ports/cli11/portfile.cmake]: the name of these options has been changed to
```
CLI11_BUILD_TESTS
CLI11_BUILD_EXAMPLES
CLI11_BUILD_DOCS
```
eg. see https://github.com/CLIUtils/CLI11/commit/60e2932356f049ba0313559c3d512bb1bc8e2b72#diff-af3b638bc2a3e6c650974192a53c7291R257

File: ports/cli11/CONTROL [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Source: cli11
-Version: 1.8.0
+Version: 1.9.0
 Description: CLI11 is a command line parser for C++11 and beyond that provides a rich feature set with a simple and intuitive interface.
 Homepage: https://github.com/CLIUtils/CLI11
+Build-Depends: gtest
```
Comment (DailyShana) [inline ports/cli11/CONTROL]: as build with test is disabled, this dependency is unnecessary

File: ports/cli11/portfile.cmake [hunk 3]
```diff
@@ -14,6 +12,8 @@ vcpkg_configure_cmake(
     OPTIONS
         -DCLI11_TESTING=OFF
         -DCLI11_EXAMPLES=OFF
```
Comment (DailyShana) [inline ports/cli11/portfile.cmake]: CLI11_TESTING is changed to CLI11_BUILD_TESTS:
remove: https://github.com/CLIUtils/CLI11/commit/60e2932356f049ba0313559c3d512bb1bc8e2b72#diff-af3b638bc2a3e6c650974192a53c7291L210
add: https://github.com/CLIUtils/CLI11/commit/60e2932356f049ba0313559c3d512bb1bc8e2b72#diff-af3b638bc2a3e6c650974192a53c7291R257
CLI11_EXAMPLES is changed to CLI11_BUILD_EXAMPLES:
remove: https://github.com/CLIUtils/CLI11/commit/60e2932356f049ba0313559c3d512bb1bc8e2b72#diff-af3b638bc2a3e6c650974192a53c7291L224
add: https://github.com/CLIUtils/CLI11/commit/60e2932356f049ba0313559c3d512bb1bc8e2b72#diff-af3b638bc2a3e6c650974192a53c7291R261

**Review by DailyShana — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
Awesome, LGTM!

**Pull request merged**